src/utils/jsonwebtoken.js

```diff
@@ -0,0 +1,124 @@
+// packages
+const jwt = require('jsonwebtoken');
+const dotenv = require('dotenv').config();
+const crypto = require('crypto');
+const fs = require('fs');
+const path = require('path');
+
+/** 
+ * @param {object} userInfo - validated user id
+ * @param {string} expirationTime - token validity
+ */
+
+/**Basic / symmetric jwt generation */
+const generatingToken = function (userInfo, expirationTime) {
+    // sample of generated payload
+    const {userId, fullName, email} = userInfo;
+    const payload = {
+        sub: userId,
```

Comment:
I believe this here is id...or what does 'sub' mean?

Reply:
i believe the 'sub' is meant for the userId claim right?

Reply:
sub is subject, it's just the convention used when generating a payload for jwt, and it's value is usually a unique id which is useful when it comes to verifying jwt. It's preferably an id or sometimes username if no id is provided. Since it's usually an id, would go with id.

Reply:
Okay subject, yeah make sense, but i think we should stick with id or userId or something like that, for clarity, if that makes sense.